src/logger.rs

```diff
@@ -138,17 +179,79 @@ impl<F: File> Drain for FileBasedDrain<F> {
         &self,
         record: &slog::Record,
         values: &slog::OwnedKVList,
-    ) -> Result<Self::Ok, Self::Err> {
-        // Ignore errors here
-        let _ = self.inner.lock().unwrap().write(
-            format!(
-                "[{}] : {:?} \n {:?} \n",
-                record.level(),
-                record.msg(),
-                values
-            )
-            .as_bytes(),
+    ) -> std::result::Result<Self::Ok, Self::Err> {
+        let by = format!(
+            "[{}] : {:?} \n {:?} \n",
+            record.level(),
+            record.msg(),
+            values
         );
+        for rotator in self.rotators.iter() {
+            if rotator.should_rotate() {
+                self.flush().unwrap();
```

Comment:
`unwrap()` seems not suitable here. At least we can throw the error I think.

Reply:
what kind of error should throw，`Error:: Corruption` ，`Error::IO` or `Error::Customized`